cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -347,28 +348,28 @@ public ApplicationWithPrograms deployApp(Id.Namespace namespace, @Nullable Strin
    * @param namespaceId the {@link Id.Namespace} under which all application should be deleted
    * @throws Exception
    */
-  public void removeAll(final Id.Namespace namespaceId) throws Exception {
+  public Iterable<Id.Program> removeAll(final Id.Namespace namespaceId) throws Exception {
```

Comment:
By returning a `Set`, I mean you can have `return ImmutableSet.copyOf(runningPrograms);`